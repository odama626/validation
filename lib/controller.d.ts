/// <reference types="react" />
export default class Controller implements Core.FormController {
    components: {
        [name: string]: Core.FormField;
    };
    Context: React.Context<Core.FormProviderContext>;
    provider: Core.Provider;
    constructor();
    attachContext(provider: any): void;
    attachComponent(name: any, component: any): void;
    validateByName(name: any, values: any): Promise<any>;
    set(name: any, opts: any): void;
    get(name: string): {
        value: any;
        valid: any;
        message: any;
    };
    /**
     * ```
     * `
     * runs ${ validate } on all connected ${ FormFields }
     * inside containing ${ Form }
     *
     * marks all ${ FormFields } invalid that return falsey
     * from ${ validate } function
     *
     * returns ${ true } if all ${ FormFields } pass validation
     * `
     * ```
     * example:
     * ```
     * if (controller.validate()) {
     *   // Form is valid, grab form data
     *   let data = controller.getValues();
     * } else {
     *  // show error message here
     * }
     * ```
     */
    validate(): Promise<any>;
    /**
     * ```
     * `
     * Get values from all connected ${ FormFields }
     *
     * returns ${ key value pairs } where the key is
     * the ${ FormField name } and the value is it's value
     * `
     * ```
     */
    getValues(): {
        [name: string]: any;
    };
}
