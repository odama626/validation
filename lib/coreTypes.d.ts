/// <reference types="react" />
export interface Validation {
    valid: boolean;
    message?: any;
}
export declare type ValidateFunc = (value: any, values: any) => Promise<boolean | Validation> | boolean | Validation;
declare type ValidationResult = Promise<Validation>;
export declare type ExtractProps<TComponentOrTProps> = TComponentOrTProps extends React.ComponentType<infer TProps> ? TProps : TComponentOrTProps;
export interface WrapperChildProps {
    value: any;
    onChange: (event: any) => any;
}
export interface FormController {
    attachComponent: (name: any, conmponent: any) => void;
    attachContext: (conmponent: any) => void;
    validate: () => ValidationResult;
    validateByName: (name: string, values: {
        [name: string]: any;
    }) => ValidationResult;
    set: (name: string, opts: {
        value?: any;
        message?: any;
        valid?: boolean;
    }) => void;
    get: (name: string) => {
        value: any;
        valid: boolean;
        message?: any;
    };
}
export interface FormField {
    validate?: ValidateFunc;
    Context: React.Context<FormProviderContext>;
}
export interface FormProviderContext {
    errors: {
        [name: string]: boolean;
    };
    values: {
        [name: string]: any;
    };
    onChange: (event: any) => void;
}
export interface Provider extends React.Component<any, any> {
    onChange: (event: React.SyntheticEvent<any>) => void;
    update: (errorFields: {
        [name: string]: any;
    }) => void;
    context: React.Context<FormProviderContext>;
}
export interface FormFieldProps<T> extends React.HTMLProps<T> {
    validate?: ValidateFunc;
    controller?: FormController;
    name: string;
}
export interface SimpleWrapperProps extends FormFieldProps<null> {
    children: React.ReactElement<any>;
}
export interface WrapperProps extends FormFieldProps<null> {
    children?: (props: WrapperChildProps & {
        name?: string;
        error: boolean;
        message?: any;
    }) => React.ReactElement<any>;
}
export interface FormProps extends React.HTMLProps<HTMLDivElement> {
    controller: FormController;
}
export {};
