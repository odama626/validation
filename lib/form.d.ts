import * as Core from './coreTypes';
import * as React from 'react';
export default class Form extends React.Component<Core.FormProps, any> implements Core.Provider {
    Context: any;
    constructor(props: any);
    onChange(event: any, directName?: string): void;
    update(errors: any): void;
    render(): JSX.Element;
}
