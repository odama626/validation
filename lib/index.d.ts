import Form from './form';
import Controller from './controller';
import Wrapper, { Wrap } from './wrapper';
import SimpleWrapper from './simpleWrapper';
import withValidation from './withValidation';
export { Form, Controller, Wrapper, SimpleWrapper, Wrap };
export default withValidation;
