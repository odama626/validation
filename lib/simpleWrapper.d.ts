import * as Core from './coreTypes';
import * as React from 'react';
/**
 * _@core / Form / Wrapper_
 *
 * ### Wrap a component that has value and onChange
 * ```
 * `
 * sets ${data-name} of component to track onChange
 * adds ${data-error} when there is an error with the input
 * `
 * ```
 * ### Accepted Props
 * ```
 * controller: Core.FormController // instance of FormController
 *
 * validate?: (value) => boolean // optional validation function
 *
 * name: string // the key the value will be under in the controller
 * ```
 */
export default class SimpleWrapper extends React.Component<Core.SimpleWrapperProps> implements Core.FormField {
    Context: React.Context<any>;
    validate(value: string, values: any): boolean | Promise<boolean>;
    componentWillMount(): void;
    render(): JSX.Element;
}
