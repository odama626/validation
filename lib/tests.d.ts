export declare const phoneNumber: () => (value: any) => true | {
    valid: boolean;
    message: string;
};
export declare const emailAddress: () => (value: any) => true | {
    valid: boolean;
    message: string;
};
export declare const notEmpty: () => (value: any) => any;
