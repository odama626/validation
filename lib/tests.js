'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var regex = {
    tel: /^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/,
    email: /^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/,
};
var phoneNumber = function () { return function (value) {
    var pass = regex.tel.test(value);
    return pass || { valid: false, message: 'not a valid phone number' };
}; };
var emailAddress = function () { return function (value) {
    return regex.email.test(value) || { valid: false, message: 'not a valid email address' };
}; };
var notEmpty = function () { return function (value) {
    return (value && value.length) || { valid: false, message: 'required' };
}; };

exports.phoneNumber = phoneNumber;
exports.emailAddress = emailAddress;
exports.notEmpty = notEmpty;
