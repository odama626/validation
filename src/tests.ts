const regex = {
  tel: /^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/,
  email: /^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/,
}

export const phoneNumber = () => value => {
  let pass = regex.tel.test(value);
  return pass || { valid: false, message: 'not a valid phone number' };
}

export const emailAddress = () => value => {
  return regex.email.test(value) || { valid: false, message: 'not a valid email address'};
}

export const notEmpty = () => value => {
  return (value && value.length) || { valid: false, message: 'required'};
}
