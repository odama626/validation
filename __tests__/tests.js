import { emailAddress, phoneNumber, notEmpty } from '../lib/tests';

describe('Provided Tests', () => {
  test('emailAddress', () => {
    expect(emailAddress()('a@b.com')).toBeTruthy();
    expect(emailAddress()('asdf')).toMatchSnapshot();
  });

  test('phoneNumber', () => {
    expect(phoneNumber()('123321123')).toMatchSnapshot();
    expect(phoneNumber()('1234566778')).toBeTruthy();
  });

  test('notEmpty', () => {
    expect(notEmpty()('')).toMatchSnapshot();
    expect(notEmpty()('blah')).toBeTruthy();
  })


})
