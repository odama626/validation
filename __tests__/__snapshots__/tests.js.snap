// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provided Tests emailAddress 1`] = `
Object {
  "message": "not a valid email address",
  "valid": false,
}
`;

exports[`Provided Tests notEmpty 1`] = `
Object {
  "message": "required",
  "valid": false,
}
`;

exports[`Provided Tests phoneNumber 1`] = `
Object {
  "message": "not a valid phone number",
  "valid": false,
}
`;
